pennylane/ops/qubit/state_preparation.py

```diff
@@ -236,6 +239,10 @@ def circuit(state=None):
         >>> state
         tensor([0.5+0.j, 0.5+0.j, 0.5+0.j, 0.5+0.j], requires_grad=True)
 
+
+    .. details::
```

Comment:
usage details added @isaacdevlugt

Reply:
Thanks @JerryChen97! I made some more edits to the docstring as well (see https://github.com/PennyLaneAI/pennylane/pull/6863/commits/4e81229ad30b5d69f85b0634da94de5d5d0e7191)

Reply:
Thanks! That's a lot more readibility